### DISCUSSION TITLE
Custom link-in-bio template

### DISCUSSION THREAD

**Discussion by JewishBidoof** [ottomated/create-o7-app#8] [Show and tell]
Hello everyone! Big thanks to Otto for this tool, he's a real one. I made a serverless link-in-bio site that uses a Postgres connection to manage links dynamically without the need to redeploy! I'm fairly new to web dev (as far as anything past standard HTML anyway) so if anyone has ways of improving it please do so and make a pull request, I'd love to see your methods!

https://github.com/TrekkerStudios/THBioLink

**Comment by tom-huntington** [comment]
> Previously I had done this with a static site, but would have to redeploy/reupload files every time I made a change. I wanted it to instead utilize a SQL-style backend to allow for it to dynamically update links.

Umm

```bash
$ git push
```
static site now redeployed...

**Reply by JewishBidoof** [reply]
Yes but that would still require me to edit the code, push the commit, wait for Vercel to redeploy, and then push it to production. I can update my db and it will immediately change on the user end. Is it a negligible time difference? Possibly. But I have multiple websites that I run with this and being able to edit everything from a single spot is useful for me. The plan is to eventually create another frontend for interacting with the database in a more user-friendly way, but I'm taking it a step at a time.

**Reply by JewishBidoof** [reply]
I realized I meant to put a bit about this in the README, so I have now done that and will copy it here:

> Eventually I want to implement a front-end for interacting with/managing the database in a user-friendly way. In essence I'm recreating the Linktree experience but in a self-hosted manner so you have full control. Is it sorta "reinventing the wheel?" Perhaps. But I'm having fun doing it so I don't care.

**Reply by tom-huntington** [reply]
Senior devs under-engineering, just saying

**Reply by tom-huntington** [reply]
I think [workers kv](https://developers.cloudflare.com/workers/runtime-apis/kv) would the proper way to achieve this dynamically. Especially since planetscale limits the number of free databases and freezes after 6 days without writes.

**Reply by JewishBidoof** [reply]
I might actually give that a look. I'm just starting to wrap my brain around edge stuff, so I hadn't really considered it before. I use Supabase so as long as it's regularly accessed it won't pause my db, however kv would make it significantly more straightforward.

**Reply by JewishBidoof** [reply]
Welp after a lot of refactoring and banging my head I did it, albeit with Vercel's KV to start, then eventually I'll figure out how to set up miniflare and get it moved over.
[Version 2.1 now up](https://github.com/TrekkerStudios/THBioLink)

**Reply by tom-huntington** [reply]
@TrekkerStudios Otto roasted your code in the latest stream https://www.twitch.tv/videos/1827386687?t=1h7m29s
